subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmPluginServices.java

```diff
@@ -283,22 +275,19 @@ public static class DefaultElementsConfigurationBuilder extends AbstractConfigur
         private boolean published;
 
         @Inject
-        public DefaultElementsConfigurationBuilder(String name, JvmPluginServices jvmEcosystemUtilities, ConfigurationContainer configurations, SoftwareComponentContainer components, TaskContainer tasks) {
-            super(name, jvmEcosystemUtilities, configurations);
+        public DefaultElementsConfigurationBuilder(String name, JvmPluginServices jvmEcosystemUtilities, ConfigurationContainerInternal configurations, SoftwareComponentContainer components, TaskContainer tasks) {
+            super(name, jvmEcosystemUtilities, (RoleBasedConfigurationContainerInternal) configurations);
             this.components = components;
             this.tasks = tasks;
         }
 
         @Override
         Configuration build() {
-            Configuration cnf = configurations.maybeCreate(name);
+            Configuration cnf = ((RoleBasedConfigurationContainerInternal) configurations).maybeCreateWithRole(name, ConfigurationRoles.INTENDED_CONSUMABLE, false, true);
```

Comment:
We're going to need to revert this or use a custom role. Since my fix for the regression merged into release, `setCanBeDeclaredAgainst(false)` has been removed.

Reply:
I suppose this PR has to wait until the next time release merges to master, so I can update this with the latest master and adjust for the removal.